src/ray/gcs/gcs_server/gcs_server.h

```diff
@@ -206,18 +207,12 @@ class GcsServer {
 
   void TryGlobalGC();
 
+  IoContextProvider<GcsServerIoContextPolicy> io_context_provider_;
```

Comment:
Should we introduce a factory method in a separate file like https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/default_impl.py

Reply:
hmm I think that'd be too elaborate since the only difference is this 1 line

Reply:
ok, let's keep as it is and revisit it if it will cause merge conflicts.